consensus/polybft/fsm.go

```diff
@@ -628,6 +628,10 @@ func getExitEventsFromReceipts(epoch, block uint64, receipts []*types.Receipt) (
 		}
 
 		for j := 0; j < len(receipts[i].Logs); j++ {
+			if receipts[i].Logs[j].Address != contracts.L2StateSenderContract {
+				continue
+			}
+
 			event, err := decodeExitEvent(convertLog(receipts[i].Logs[j]), epoch, block)
```

Comment:
Although not sure if it makes more sense adding this check inside `decodeExitEvent` itself. WDYT?

```suggestion
                        log := receipts[i].Logs[j]
			if log.Address != contracts.L2StateSenderContract {
				continue
			}

			event, err := decodeExitEvent(convertLog(log), epoch, block)
```

Reply:
We don't need to decode events not from L2StateSenderContract

Reply:
I know, I was just suggesting moving that check to `decodeExitEvent` function as well.

However still would like to create a local variable `log` instead of indexing slices twice.

Reply:
Switched to for range. My idea was to filter not relevant logs as early as possible. Because if we move this check to decode function we'll have to convert every log to ethgo format, before filtering.

Reply:
Ok makes sense :wink: